stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/factory/StreamViewModelFactory.kt

```diff
@@ -0,0 +1,32 @@
+package com.getstream.sdk.chat.viewmodel.factory
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.getstream.sdk.chat.viewmodel.ChannelHeaderViewModel
+import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
+import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel
+
+/**
+ * A ViewModel factory for MessageListViewModel, ChannelHeaderViewModel and MessageInputViewModel
+ *
+ * @param cid: the channel id in the format messaging:123
+ *
+ * @see MessageListViewModel
+ * @see ChannelHeaderViewModel
+ * @see MessageInputViewModel
+ */
+public class StreamViewModelFactory(private val cid: String) : ViewModelProvider.Factory {
```

Comment:
Not sure about the naming of this class yet, but we can't have `ChannelsViewModelFactory` and `ChannelViewModelFactory` in the API, way too confusing. Should review this with the things discussed [here](https://getstream.slack.com/archives/G016YAWD423/p1603467611056200).

Reply:
Maybe `ChannelListViewModelFactory` and `MessageListViewModelFactory`?